parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -950,7 +958,7 @@ protected PageHeader readPageHeader() throws IOException {
         // to allow reading older files (using dictionary) we need this.
         // usually 13 to 19 bytes are missing
         // if the last page is smaller than this, the page header itself is truncated in the buffer.
-        this.byteBuf.rewind(); // resetting the buffer to the position before we got the error
+        this.byteBuf.position(initialPos); // resetting the buffer to the position before we got the error
```

Comment:
I'm not sure this code is even needed any more. The WorkaroundChunk class was added 3 years back and the comment from then says "it deals with a now fixed bug where compressedLength was missing a few bytes". While we do seem to exercise it (as we always use it for the last chunk), I'm not sure how often the bug which this catch was meant to solve gets hit. 
If we need it, I can try and create a unit test for the WorkaroundChunk class. Don't see a straightforward way to test it as part of the FileReader as it is too many levels deep in the class.